### DISCUSSION TITLE
Tapping (not clicking) with two fingers triggers the iPadOS context menu instead of the vs code context menu

### DISCUSSION THREAD

**Discussion by tonyliangli** [blinksh/blink#1846] [Support]
### Checklist

- [x] Using latest version on the [App Store](http://itunes.apple.com/app/id1156707581)
- [x] Read the [docs](http://blink.sh/docs/)
- [x] Searched for [existing GitHub issues](https://github.com/blinksh/blink/issues)

### Configuration

Blink 16.2.0 TF version

### Describe the bug

When the follow setting for iPadOS is turned on: Settings->General->Trackpad->Tap to Click & Two-finger Secondary Click, the context menu inside code is not working properly. There will be two context menus popping out when you “tap” on the touchpad with two fingers. Instead, when you “click” on the touchpad with two fingers, only the vs code context menu will pop out.

Only the vs code context menu should pop out when “tapping” on the touchpad with two fingers.

![IMG_0054](https://github.com/blinksh/blink/assets/16441239/5e5572b6-d130-4bca-b772-db335304cb9d)

**Comment by thekeith** [comment]
+1 to this issue, currently occurring as well in iPadOS 17 beta 6 on the testflight version of Blink Shell

**Comment by thekeith** [comment]
@tonyliangli are you on iPadOS 17 beta or iPadOS 16?

**Comment by tonyliangli** [comment]
@thekeith Both won’t work. @yury Can this problem possibly be fixed?

**Comment by thekeith** [comment]
@tonyliangli [I've reported the issue to Webkit](https://bugs.webkit.org/show_bug.cgi?id=260722). So far we're still homing in on the problem, but it seems to be a difference in how tap-based events are handled with a trackpad in iPadOS that triggers it. It also reproduces in iPadOS 16.6+, so this problem has been going on for a while.

A workaround may be having a separate handler for [the uitapgesturerecognizer class](https://developer.apple.com/documentation/uikit/uitapgesturerecognizer) but that is only speculation at this point.

**Comment by tonyliangli** [comment]
> @tonyliangli [I've reported the issue to Webkit](https://bugs.webkit.org/show_bug.cgi?id=260722). So far we're still homing in on the problem, but it seems to be a difference in how tap-based events are handled with a trackpad in iPadOS that triggers it. It also reproduces in iPadOS 16.6+, so this problem has been going on for a while.
> 
> 
> 
> A workaround may be having a separate handler for [the uitapgesturerecognizer class](https://developer.apple.com/documentation/uikit/uitapgesturerecognizer) but that is only speculation at this point.

Understood. Thanks for the reply.

**Comment by carloscabanero** [comment]
Hi! We discussed this during our product meeting. I actually never noticed this myself as I always did the "two finger click" instead of "two finger tap" in the trackpad. 

We won't be prioritizing this as it is really minor (clicking is what most users do), it looks like it is more an issue with WebKit, and we have bigger fish to fry for Blink Code. We think we need to put our very scarce resources on other things, like releasing Blink-FS as open source instead.

Will move this to a Discussion, so whenever this gets fixed on WebKit we will all find out. If in the future we have more support we can work on it. But please let us know if you think otherwise.

**Comment by thekeith** [comment]
@carloscabanero there may be a very simple fix for this, related to handling the two-finger-tap in UITapGestureRecognizer. It looks like both the tap and the right click propagate through to the DOM, so intercepting the tap may work.
